### ISSUE TITLE
ERROR: No matching issuer found

### ISSUE THREAD

**Issue by abiansyahn** [MRHRTZ/Shopee-flashsale-bot#6]
ini kenapa ya om, error pas isi shopee url-nya

```
Checking url ...


DevTools listening on ws://127.0.0.1:58733/devtools/browser/57f69ab4-6b08-4768-a2f7-d87d53979c92
[0405/225852.626:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for shopee.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225857.161:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for deo.shopeemobile.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225900.420:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for www.googletagmanager.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225901.196:INFO:CONSOLE(1)] "A preload for 'https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/ProductDetailsPage.954d5bb98220d4f7.js' is found, but is not used because the request credentials mode does not match. Consider taking a look at crossorigin attribute.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/webpack-runtime.516ea1d87588319d.js (1)
[0405/225902.937:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for o468184.ingest.sentry.io failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225903.508:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for dem.shopee.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225904.866:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for analytics.tiktok.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225904.966:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for connect.facebook.net failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225907.070:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for 9469210.fls.doubleclick.net failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225908.375:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for www.google-analytics.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225909.070:INFO:CONSOLE(1)] "[Meta Pixel] - Duplicate Pixel ID: 466924370133774.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/3849.a55c3fa99447f350.js (1)
[0405/225911.446:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for www.google.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225911.936:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for adservice.google.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225912.866:INFO:CONSOLE(0)] "[.WebGL-000012BC033A2A00]GL Driver Message (OpenGL, Performance, GL_CLOSE_PATH_NV, High): GPU stall due to ReadPixels", source: https://shopee.co.id/Mito-Air-Fryer-AF8-Crystal-Series-i.24544583.21047214417https://shopee.co.id/Mito-Air-Fryer-AF8-Crystal-Series-i.24544583.21047214417 (0)
[0405/225913.534:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for ampcid.google.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225913.750:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for www.facebook.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225914.199:INFO:CONSOLE(1)] "There is already an IndexedDBStorage / memostorage in session manager, can not update storage.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/3849.a55c3fa99447f350.js (1)
[0405/225916.667:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for www.google.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225920.205:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for df.infra.sz.shopee.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225920.285:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for ampcid.google.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225921.139:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for adservice.google.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225921.657:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for firebaseinstallations.googleapis.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225922.556:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for analytics.google.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225923.649:INFO:CONSOLE(1)] "A preload for 'https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/6156.32aeb353c6f62bdd.js' is found, but is not used because the request credentials mode does not match. Consider taking a look at crossorigin attribute.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/webpack-runtime.516ea1d87588319d.js (1)
[0405/225923.681:INFO:CONSOLE(1)] "A preload for 'https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/PageAntiCrawlerVerify.dad5fb9d2a6a5279.js' is found, but is not used because the request credentials mode does not match. Consider taking a look at crossorigin attribute.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/webpack-runtime.516ea1d87588319d.js (1)
[0405/225923.919:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for googleads.g.doubleclick.net failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225926.249:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for stats.g.doubleclick.net failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[0405/225928.657:INFO:CONSOLE(1)] "There is already an IndexedDBStorage / memostorage in session manager, can not update storage.", source: https://deo.shopeemobile.com/shopee/shopee-pcmall-live-sg/assets/3849.a55c3fa99447f350.js (1)
[0405/225929.146:ERROR:web_contents_delegate.cc(246)] WebContentsDelegate::CheckMediaAccessPermission: Not supported.
[0405/225929.169:ERROR:web_contents_delegate.cc(246)] WebContentsDelegate::CheckMediaAccessPermission: Not supported.
[0405/225930.516:INFO:CONSOLE(0)] "[.WebGL-000012BC033A2A00]GL Driver Message (OpenGL, Performance, GL_CLOSE_PATH_NV, High): GPU stall due to ReadPixels", source: https://shopee.co.id/verify/captcha?anti_bot_tracking_id=d025365e-e65d-4544-aadd-6be037699446&app_key=PDP.PC&client_id=1&next=https%3A%2F%2Fshopee.co.id%2Fverify%2Ftraffic&redirect_type=2&scene=crawler_item&should_replace_history=true (0)

 [ Item not found ]

Invalid URL!
```

**Comment by tahaluindo** [comment]
Invalid URL!

**Comment by MRHRTZ** [comment]
First for certificate error, download [ca.crt](https://raw.githubusercontent.com/MRHRTZ/Shopee-flashsale-bot/master/config/ca.crt) then import to your chrome `click Privacy and security > Click Security > Scroll to Advanced > Click Manage certificates` .

And for item not found this caused by a recaptcha login which has been fixed in the latest update [ad5fbf6](https://github.com/MRHRTZ/Shopee-flashsale-bot/commit/ad5fbf62e7d60421dcef6014266634e68a536097)

**Comment by hutomotrih** [comment]
> Pertama buat certificate error, download file [ca.crt](https://raw.githubusercontent.com/MRHRTZ/Shopee-flashsale-bot/master/config/ca.crt) Terus import ke chrome `click Privacy and security > Click Security > Scroll to Advanced > Click Manage certificates` .
> 
> Dan juga masalah item not found itu karna masalah recaptcha login, sudah di fix di update terbaru [ad5fbf6](https://github.com/MRHRTZ/Shopee-flashsale-bot/commit/ad5fbf62e7d60421dcef6014266634e68a536097)

error bang

` [ Sign in process ... ]
  [ Loading account ... ]

[6412:17236:0406/161753.612:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for c-api-bit.shopeemobile.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:10740:0406/161755.743:ERROR:device_event_log_impl.cc(222)] [16:17:55.743] USB: usb_device_handle_win.cc:1046 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
    challenge_type: 1
    captcha: {'scene': 'crawler_item'}
    captcha_extra_info: {'tracking_id': '105dafd1-c0e5-433b-809a-93cde88fb59f'}
    action_type: 3
    error: 90309999
    is_customized: False
    is_login: False
    platform: 0
    report_extra_info: 

  [ Item not found, This can happen because login error, attempting second phase in 10 sec ... ]

[6412:17236:0406/161802.944:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller.shopee.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:17236:0406/161806.930:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller.shopee.sg failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:17236:0406/161807.186:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for optimizationguide-pa.googleapis.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:17236:0406/161810.949:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for chatbot.shopee.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:17236:0406/161811.468:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller-push-ws.shopee.co.id failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found


[6412:17236:0406/161811.716:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for firebaseremoteconfig.googleapis.com failed:
----- Certificate i=1 (CN=Selenium Wire CA) -----
ERROR: No matching issuer found



  [ Second phase failed! ]`

**Comment by MRHRTZ** [comment]
> > Pertama buat certificate error, download file [ca.crt](https://raw.githubusercontent.com/MRHRTZ/Shopee-flashsale-bot/master/config/ca.crt) Terus import ke chrome `click Privacy and security > Click Security > Scroll to Advanced > Click Manage certificates` .
> > Dan juga masalah item not found itu karna masalah recaptcha login, sudah di fix di update terbaru [ad5fbf6](https://github.com/MRHRTZ/Shopee-flashsale-bot/commit/ad5fbf62e7d60421dcef6014266634e68a536097)
> 
> error bang
> 
> ` [ Sign in process ... ] [ Loading account ... ]
> 
> [6412:17236:0406/161753.612:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for c-api-bit.shopeemobile.com failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:10740:0406/161755.743:ERROR:device_event_log_impl.cc(222)] [16:17:55.743] USB: usb_device_handle_win.cc:1046 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F) challenge_type: 1 captcha: {'scene': 'crawler_item'} captcha_extra_info: {'tracking_id': '105dafd1-c0e5-433b-809a-93cde88fb59f'} action_type: 3 error: 90309999 is_customized: False is_login: False platform: 0 report_extra_info:
> 
> [ Item not found, This can happen because login error, attempting second phase in 10 sec ... ]
> 
> [6412:17236:0406/161802.944:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller.shopee.co.id failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:17236:0406/161806.930:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller.shopee.sg failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:17236:0406/161807.186:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for optimizationguide-pa.googleapis.com failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:17236:0406/161810.949:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for chatbot.shopee.co.id failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:17236:0406/161811.468:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for seller-push-ws.shopee.co.id failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [6412:17236:0406/161811.716:ERROR:cert_verify_proc_builtin.cc(677)] CertVerifyProcBuiltin for firebaseremoteconfig.googleapis.com failed: ----- Certificate i=1 (CN=Selenium Wire CA) ----- ERROR: No matching issuer found
> 
> [ Second phase failed! ]`

Import dulu certificates ke chrome biar gak failed certificate (optional) 
Problem second phase karna timeout udah di fix [a9afefb](https://github.com/MRHRTZ/Shopee-flashsale-bot/commit/a9afefb460f8ad6a59d65e6217e9d66a3bda178b)

**Comment by hutomotrih** [comment]
ada error saat flash sale di mulai min

" [ FlashSale Started. ]
  [ we have an error, please open issue on github. ]

Traceback (most recent call last):
  File "C:\Users\hutom\Downloads\Shopee-flashsale-bot-master\Shopee-flashsale-bot-master\lib\driverExecutor.py", line 188, in executeScript
    to_cart = driver.find_element(by=By.CSS_SELECTOR, value=to_cart_active)
  File "C:\Users\hutom\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 831, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
  File "C:\Users\hutom\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "C:\Users\hutom\AppData\Local\Programs\Python\Python310\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"button.btn.btn-solid-primary.btn--l"}
  (Session info: chrome=112.0.5615.50)
Stacktrace:
Backtrace:
        GetHandleVerifier [0x009CDCE3+50899]
        (No symbol) [0x0095E111]
        (No symbol) [0x00865588]
        (No symbol) [0x008908F9]
        (No symbol) [0x00890AFB]
        (No symbol) [0x008BF902]
        (No symbol) [0x008AB944]
        (No symbol) [0x008BE01C]
        (No symbol) [0x008AB6F6]
        (No symbol) [0x00887708]
        (No symbol) [0x0088886D]
        GetHandleVerifier [0x00C33EAE+2566302]
        GetHandleVerifier [0x00C692B1+2784417]
        GetHandleVerifier [0x00C6327C+2759788]
        GetHandleVerifier [0x00A65740+672048]
        (No symbol) [0x00968872]
        (No symbol) [0x009641C8]
        (No symbol) [0x009642AB]
        (No symbol) [0x009571B7]
        BaseThreadInitThunk [0x776B7D69+25]
        RtlInitializeExceptionChain [0x77CFB74B+107]
        RtlClearBits [0x77CFB6CF+191]


  [ Press Anykey To Exit This Script ]"
